packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -9,15 +9,39 @@
 from jupyter_ai.config_manager import ConfigManager, Logger
 from jupyter_ai.models import AgentChatMessage, HumanChatMessage
 from jupyter_ai_magics.providers import BaseProvider
+from traitlets.config import Configurable
 
 if TYPE_CHECKING:
     from jupyter_ai.handlers import RootChatHandler
 
 
-class BaseChatHandler:
+class BaseChatHandler(Configurable):
     """Base ChatHandler class containing shared methods and attributes used by
     multiple chat handler classes."""
 
+    # Class attributes
+    id: str = "base-chat-handler"
+    """ID for this chat handler; should be unique"""
+
+    name: str = "Base Chat Handler"  # TODO: make NotImplemented
+    """User-facing name of this handler"""
+
+    description: str = (
+        "Handler for messages that are not commands"  # TODO: make NotImplemented
+    )
+    """Description used for routing requests, to be used when dispatching
+    messages to model providers. Also shown in the UI."""
+
+    # TODO: make NotImplemented
+    help: str = "This is used when the message in the chat interface is not a command"
+    """What this chat handler does, which third-party models it contacts,
+    the format of the data it returns to the user, etc. Used in the UI."""
```

Comment:
What is the `# TODO: make NotImplemented` comment meant to indicate? If it doesn't make sense to define these on the base class, you can use an ellipsis to indicate that subclasses should override this attribute:

```
name: ClassVar[str] = ... 
```

Sadly Pydantic, despite recommending this strategy, does not actually enforce subclasses to implement ellipsis-defined class attributes. So enforcing this would have to be done by a custom Python test. See: https://github.com/pydantic/pydantic/issues/4448